### PR TITLE
Fix memory usage for findAllDeps

### PR DESCRIPTION
**Pull request by eeue56**
Reduces memory usage by a huge amount. Also fixes some deps tracking upstream for webpack-loader

### CODE REVIEW

File: dependencies.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            if (line.indexOf('import ') === 0){
+                imports.push(line);
+            } else if (
+                line.indexOf(' ') === 0
+                || line.trim().length === 0
+                || line.startsWith('--')
+                || line.startsWith('{-')
+                || line.startsWith('-}')
+                ) {
+                // ignore lines starting with whitespace while parsing imports
```
Comment (stoeffel) [inline dependencies.js]: I don't have much context on how this is used, but I think this doesn't work with multiline imports, right?

Comment (rtfeldman) [inline dependencies.js]: Also worth noting:

```elm
import Foo

    {- I am a block comment wth whitespace first
import NotRealBecauseStillInABlockComment
                 -}

import RealImport
```

Comment (eeue56) [inline dependencies.js]: We don't need to grab multiline imports

**Comment by eeue56** [comment]
- Reduces number of FD needed
- Parses in a more reliable way
- Stop reading all the files when not needed and stop using fs-stat when we are doing a read anyway

**Comment by eeue56** [comment]
tests pass locally. not sure why they don't pass on travis - but it's a function completely unrelated to what I've changed. merging in a minute

**Pull request merged**